index.html

```diff
@@ -562,8 +563,120 @@ <h2>Dictionary {{RTCRtpEncodingParameters}} Members</h2>
           allows the <a>user agent</a> to adapt its bandwidth allocation strategy
           based on the current network conditions.</p>
         </dd>
+        <dt>
+          <dfn data-idl>codec</dfn> of type <span class="idlMemberType">RTCRtpCodec</span>
+        </dt>
+        <dd>
+          <p>Optional value overriding the codec used for this encoding. The {{RTCRtpCodec}} dictionary is
+            defined in [[WEBRTC]]. If absent, the default codec selection algorithm selects the first
+            negotiated codec.</p>
+        </dd>
       </dl>
     </section>
+    <section id="rtcrtpencodingparameters-modifications">
+      <h2>Modifications to existing procedures</h2>
+      <h3>setParameters()</h3>
+      <p>Add the following steps in the parameter validation for {{RTCRtpSender/setParameters()}},
+        before any validation of <var>scalabilityMode</var> from [[WEBRTC-SVC]]:</p>
+      <ol>
+        <li><p>Let <var>codecs</var> be <var> parameters</var>.{{RTCRtpParameters/codecs}}.</p>
```

Comment:
Hmm, webrtc-pc already contains this (unused!) line, so assuming https://github.com/w3c/webrtc-pc/pull/2854 gets merged, how about something like: _"Under the [=setParameters validation steps=]"_, replace this line:

 1. Let codecs be parameters.[codecs](https://w3c.github.io/webrtc-pc/#dom-rtcrtpparameters-codecs).

with:

 1. Let _choosableCodecs_ be parameters.[codecs](https://w3c.github.io/webrtc-pc/#dom-rtcrtpparameters-codecs).
 2. If _choosableCodecs_ is an empty list, set _choosableCodecs_ to <var>transceiver</var>.{{RTCRtpTransceiver/[[PreferredCodecs]]}}
 3. If _choosableCodecs_ is still an empty list, set _choosableCodecs_ to the [=[list of implemented send codecs](https://w3c.github.io/webrtc-pc/#dfn-list-of-implemented-send-codecs)=] for [=[transceiver kind](https://w3c.github.io/webrtc-pc/#dfn-transceiver-kind)=]

And then follow SVC's lead above and say: _"Add the following to the conditions under which the operation causes a promise rejected with an [InvalidModificationError](https://webidl.spec.whatwg.org/#invalidmodificationerror) (step 4 within step 6):"_
- Any _encoding_ in encodings [=map/exists|contains=] a codec [= codec match | not found =] in choosableCodecs

This should avoid the if/else/else and hopefully be readable. WDYT?

Reply:
To precent attempts to set illegal combinations of codecs and `scalabilityMode` values, I think you need to also set the "most preferred codec" slot and refer to this slot in WebRTC-SVC Section 4.2.2.